apps/admin-portal/src/components/identity-providers/settings/attribute-management/role-mapping-settings.tsx

```diff
@@ -0,0 +1,147 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AlertLevels } from "@wso2is/core/dist/src/models";
+import { addAlert } from "@wso2is/core/dist/src/store";
+import { Heading, Hint } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { Divider, Grid } from "semantic-ui-react";
+import { getRolesList } from "../../../../api";
+import { IdentityProviderRoleMappingInterface, RoleListInterface, RolesInterface } from "../../../../models";
+import { DynamicField } from "../../../claims";
+
+/**
+ * Proptypes for the identity providers settings component.
+ */
+interface RoleMappingSettingsPropsInterface {
+
+    /**
+     * Trigger submission.
+     */
+    triggerSubmit: boolean;
+
+    /**
+     *  Handle submission.
+     */
+    onSubmit?: (roleMappings: IdentityProviderRoleMappingInterface[]) => void;
+
+    /**
+     * Roles of the IDP
+     */
+    initialRoleMappings?: IdentityProviderRoleMappingInterface[];
+}
+
+/**
+ *  Identity Provider and advance settings component.
+ *
+ * @param {IdentityProviderSettingsPropsInterface} props - Props injected to the component.
+ * @return {ReactElement}
+ */
+export const RoleMappingSettings: FunctionComponent<RoleMappingSettingsPropsInterface> = (
+    props: RoleMappingSettingsPropsInterface
+): ReactElement => {
+
+    const [roleList, setRoleList] = useState<RolesInterface[]>();
+
+    const {
+        onSubmit,
+        triggerSubmit,
+        initialRoleMappings
+    } = props;
+
+    const dispatch = useDispatch();
+
+    /**
+     * Filter out Application related and Internal roles
+     */
+    const getFilteredRoles = () => {
+        const filterRole: RolesInterface[] = roleList.filter(
+            (role) => {
+                return !(role.displayName.includes("Application/") || role.displayName.includes("Internal/"))
+            });
+
+        return filterRole.map(role => {
+            return {
+                id: role.displayName,
+                value: role.displayName
+            }
+        });
+    };
+
+    useEffect(() => {
+        getRolesList(null)
+            .then((response) => {
+                if (response.status === 200) {
+                    const allRole: RoleListInterface = response.data;
+                    setRoleList(allRole.Resources);
+                }
+            })
+            .catch(() => {
```

Comment:
Enhance the catch statement. Please check other places as well.

Reply:
Fixed in 73f0cc02c52d71ca5acc51cf4120b806799a9d3d